app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestList.java

```diff
@@ -39,4 +45,57 @@ public void replaceList( RequestList newList ) {
         this.addAll( newList );
         notifyListeners();
     }
+
+    /**
+     * Appends new requests to the end of the list and notifies listeners of the update. This method
+     * includes a limit on the size of the ArrayList.
+     * @param requests The RequestList to append to this RequestList
+     * @param maxSize The maximum length of the ArrayList
+     */
+    public void append(RequestList requests, int maxSize) {
+        for(Request request : requests) {
+            if(!this.contains(request.getId()) && this.size() <  maxSize) {
+                // still room on list and the request is not already in the list
+                this.add(request);
+            } else if(!this.contains(request.getId())) {
```

Comment:
This code should work. I don't get why I was getting that bug. It is detailed in the comments section of your pull request